packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -165,13 +165,15 @@ const DatePickerInput = (props) => {
   const [dayInitialFocused, setDayInitialFocused] = useState(false);
   const [monthInitialFocused, setMonthInitialFocused] = useState(false);
   const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  // Added below state for invalid input message for SR
+  const [invalidInput, SetInvalidInput] = useState(false);
```

Comment:
I think we already have `isInvalid` prop for date-picker. so we should make use of this prop to identify the invalid state of the date-picker component.

Reply:
This isInvalid Prop is used for css style internally. But we added this invalidInput for screen reader when they enter character.

Reply:
I don't see any usage of the state `invalidInput` in DateInput. it's just being set to true and false

Reply:
we are validating on every change triggered and if it is not a valid character we use this state to set a text and toggle the state for the inner text on visually hidden element.

Reply:
you are calling this function only when there is invalid input. not being called when the valid character is entered. In this case you don't need the state variable as well as the parameter in function

Reply:
Updated the handle function.
<img width="853" alt="Screenshot 2023-09-08 at 5 43 08 PM" src="https://github.com/cerner/terra-framework/assets/132886427/3b0a073e-6041-4793-b8a5-6ee8bc3c814a">
And we are toggling to false on valid change.
<img width="723" alt="Screenshot 2023-09-08 at 5 44 52 PM" src="https://github.com/cerner/terra-framework/assets/132886427/0c5331f0-cd36-43f3-a2bb-3e1b30e95428">

Reply:
Removed state dependency on Invalid key entry. Thanks